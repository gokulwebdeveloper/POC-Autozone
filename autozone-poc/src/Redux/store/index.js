import {createStore, applyMiddleware, compose} from 'redux';
import rootReducer from '../reducers';
import rootSaga from '../middlewares/saga';
import createSagaMiddleware from 'redux-saga';

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();

//list of middlewares
const middlewares = [sagaMiddleware];

//apply middleware on store
const store = createStore(rootReducer, applyMiddleware(...middlewares));

//run saga generator function
sagaMiddleware.run(rootSaga);

export default store;