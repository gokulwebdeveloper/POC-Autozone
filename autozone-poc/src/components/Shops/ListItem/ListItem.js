import React, { Fragment } from "react";
import PropTypes from 'prop-types';

ListItem.propTypes = {
    items: PropTypes.string,
    items: PropTypes.any.isRequired,
}

// Specifies the default values for props:
ListItem.defaultProps = {
    items: ['All']
};

export default function ListItem(props) {


    if (props != undefined && props.items != undefined) {
        return (
            <Fragment>
                {
                    props.items.map((item, key) => {
                        return <li key={item}><a href="#">{item}</a></li>
                    }
                    )}
            </Fragment>
        );
    } else {
        return <p>no data</p>
    }


}