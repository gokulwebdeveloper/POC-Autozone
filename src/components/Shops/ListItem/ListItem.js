import React, { Fragment } from "react";
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { changeStyle } from '../ShopHelper.js'
import { filterProducts } from '../../../Redux/actions';

ListItem.propTypes = {
    items: PropTypes.string,
    items: PropTypes.any.isRequired,
}

// Specifies the default values for props:
ListItem.defaultProps = {
    items: ['All']
};

function ListItem(props) {
    function handleClick(e) {
        e.preventDefault();
        changeStyle(e);
        props.filterProducts(props.productData, e.currentTarget.dataset.id, e.currentTarget.dataset.title);

    }

    if (props != undefined && props.items != undefined) {
        return (
            <Fragment>
                {

                    props.items.map((item, key) => {
                        return <li key={item} data-id={item}
                            data-title={props.title} onClick={handleClick}>
                            {item}
                        </li>
                    })
                }
            </Fragment>
        );
    } else {
        return <p>no data</p>
    }
}

const mapStateToProps = ({ productData }) => ({ productData })
export default connect(mapStateToProps, { filterProducts })(ListItem);