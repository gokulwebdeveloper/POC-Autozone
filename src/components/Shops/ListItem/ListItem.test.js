import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ListItem from './ListItem';
import store from '../../../Redux/store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });


describe('Brand testing', () => {
    var categoryWithSubmenu = {
        "title": "Brakes and Traction Control",
        "items": [
            "All",
            "Brake Rotor",
            "Brake Power Booster",
            "Hoodies Sweats",
            "Brake Shoes Rear",
            "Wheel Cylinder Rear"
        ]
    };

    beforeEach(() => {

    });

    let wrapper = mount(<Provider store={store}><ListItem items={categoryWithSubmenu.items} title={categoryWithSubmenu.title} /></Provider>);

    it("SnapShot Match for ListItem component", () => {
        const page = renderer.create(<Provider store={store}><ListItem items={categoryWithSubmenu.items} title={categoryWithSubmenu.title} /></Provider>).toJSON();
        expect(page).toMatchSnapshot();
    });

    it("li value in ListItem", () => {
        expect(wrapper.find("li").first().text()).toBe("All");
        expect(wrapper.find("li").at(1).text()).toBe("Brake Rotor");
        expect(wrapper.find("li").at(2).text()).toBe("Brake Power Booster");
        expect(wrapper.find("li").at(3).text()).toBe("Hoodies Sweats")
            ;
        expect(wrapper.find("li").at(4).text()).toBe("Brake Shoes Rear");
    });

})