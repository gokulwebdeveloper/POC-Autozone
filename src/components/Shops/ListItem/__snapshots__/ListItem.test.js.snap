// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand testing SnapShot Match for ListItem component 1`] = `
Array [
  <li
    data-id="All"
    data-title="Brakes and Traction Control"
    onClick={[Function]}
  >
    All
  </li>,
  <li
    data-id="Brake Rotor"
    data-title="Brakes and Traction Control"
    onClick={[Function]}
  >
    Brake Rotor
  </li>,
  <li
    data-id="Brake Power Booster"
    data-title="Brakes and Traction Control"
    onClick={[Function]}
  >
    Brake Power Booster
  </li>,
  <li
    data-id="Hoodies Sweats"
    data-title="Brakes and Traction Control"
    onClick={[Function]}
  >
    Hoodies Sweats
  </li>,
  <li
    data-id="Brake Shoes Rear"
    data-title="Brakes and Traction Control"
    onClick={[Function]}
  >
    Brake Shoes Rear
  </li>,
  <li
    data-id="Wheel Cylinder Rear"
    data-title="Brakes and Traction Control"
    onClick={[Function]}
  >
    Wheel Cylinder Rear
  </li>,
]
`;
